### PR TITLE
Add Mock Command

### PR DESCRIPTION
**Pull request by DLMousey**
This PR adds a `!mock` command to the bot.

If you spot a user saying something that needs a bit of sArCaSm adding to it, don't go wearing out your shift key typing it by hand, because ConsoleApp1 comes with a command built in to do exactly that.

Simply call `!mock` and mention the user you want to mock (eg. `!mock @DLMousey`), the bot will then go retrieve the last text message the user sent to the channel, and apply a little sArCaSm.
Here's a screenshot of the bot in action;
image

If you fail to provide a username, the bot will insult you and mock the last message you sent before the `!mock` call. Something we discovered during testing was that even code blocks can't escape getting the sarcastic treatment;
 
image

### CODE REVIEW

**Comment by jacksonisiah** [comment]
Leaving this here (we don't have to do this now) but insults should be unhardcoded and databased as discussed in the dev discord :owl:

**Comment by Tayo-00** [comment]
New changes came through related to the interface and a base class. Review the ping command for the new implementation.

When this is merged, please create an issue for the database part of insulting (like @jacksonisiah mentioned). This depends on another issue, being the actual database connection.

File: Modules/Commands/Mock.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                    lastMessageParts[i] = char.ToUpper(lastMessageParts[i]);
+                }
+                else
+                {
+                    lastMessageParts[i] = char.ToLower(lastMessageParts[i]);
+                }
+            }
+            
+            string newMessage = new string(lastMessageParts);
+            await ctx.TriggerTypingAsync();
```
Comment (Tayo-00) [inline Modules/Commands/Mock.cs]: I'd trigger this earlier (or not at all). When it starts mutating a message it should start "typing".

File: Modules/Commands/Mock.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            
+            if (member == ctx.Member)
+            {
+                await ctx.RespondAsync(member.Mention + " You didn't give me a username to mock, you " + randomInsult(true));
```
Comment (Tayo-00) [inline Modules/Commands/Mock.cs]: Use string interpolation. Looks better.

Comment (Tayo-00) [inline Modules/Commands/Mock.cs]: Why does `randomInsult()` always get true as parameter? Why not just remove the parameter if it's always the same?
We'll end up with a database for insults anyway, so I don't think doing anything more than this is worth the time imo.

Comment (DLMousey) [reply Modules/Commands/Mock.cs]: This whole method is probably going to be removed anyway, I'll hard code the response for the time being and add a method for retrieving `randomInsult` from the DB once that's in place.

Comment (Tayo-00) [reply Modules/Commands/Mock.cs]: How do you want to go about it. Leave this be for now and create an issue for it?

Comment (DLMousey) [reply Modules/Commands/Mock.cs]: I'll fit it up tonight and update this PR, might as well keep it all in one place

File: Modules/Commands/Mock.cs [hunk 3]
```diff
@@ -50,16 +50,16 @@ public async Task Command(CommandContext ctx, DiscordMember member = null)
                 }
             }
             
-            string newMessage = new string(lastMessageParts);
-            if (member == ctx.Member)
+            if (missingTargetUser)
             {
-                await ctx.RespondAsync($"{ctx.Member.Mention} you didn't give a username to mock, you idiot.");
+                await ctx.RespondAsync($"{member.Mention} you didn't give you a username, you moron");
```
Comment (Tayo-00) [inline Modules/Commands/Mock.cs]: you didn't give you a username, you moron
Remove the second you and it's ready for merging

**Comment by Tayo-00** [comment]
Random insult can now by done through the database. Use word type id 2.

**Pull request merged**